### PR TITLE
Create PerformanceProfile and wait for MCP updated in test code

### PR DESCRIPTION
**Pull request by slintes**
In order to be able to provide a test binary for CNF compliance tests, we want to configure the cluster in test code, before running the actual tests.

- renamed `worker-rt` to `worker-cnf` everywhere: this reflects that the realtime kernel is only 1 of many features of the operator, and aligns with the feature deploy repository. The actual role name is still configurable, but heads up, the env var name changed to `ROLE_WORKER_CNF`!
- also made the default profile name "performance" (still configurable with `PERF_TEST_PROFILE`)
- left an example profile manifest in a new kustomize dir for easier local tests and deployments
- some code cleanup and refactorings

@fedepaol: do we want to create the MCP in test code as well? If so, here or in the test suite in the feature deploy repo? (My opinion: leave it to the cluster admin, same for labelling the nodes)

/cc @dshchedr

### CODE REVIEW

**Comment by slintes** [comment]
/hold

let #170 merge first, I don't want it to need to be rebased again

**Comment by fedepaol** [comment]
> @fedepaol: do we want to create the MCP in test code as well? If so, here or in the test suite in the feature deploy repo? (My opinion: leave it to the cluster admin, same for labelling the nodes)
> 
> /cc @dshchedr

It makes sense to leave it to the cluster admin, also because other tests also depend on it.

File: functests/0_config/test_suite_performance_config_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package __performance_config_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"testing"
+
+	ginkgo_reporters "kubevirt.io/qe-tools/pkg/ginkgo-reporters"
+
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/junit"
```
Comment (fedepaol) [inline functests/0_config/test_suite_performance_config_test.go:12]: You did not import the ` __performance_config` here, so it won't be executed.

File: functests/utils/mcps/mcps.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	}
+	return mcps.Items, nil
+}
+
+// GetByName returns the MCP with the specified name
+func GetByName(name string) (*machineconfigv1.MachineConfigPool, error) {
+	mcp := &machineconfigv1.MachineConfigPool{}
+	key := types.NamespacedName{
+		Name:      name,
+		Namespace: "",
```
Comment (fedepaol) [inline functests/utils/mcps/mcps.go:49]: Just a nit, but no need to specify Namespace if it's empty string.

**Comment by fedepaol** [comment]
Another questions:
here we run tests with 

`GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --keepGoing -requireSuite functests -- -junitDir /tmp/artifacts`

Will it catch also the new suite?

File: functests/0_config/test_suite_performance_config_test.go (hunk 1 above)
Comment (slintes) [reply functests/0_config/test_suite_performance_config_test.go:12]: Seems it is not needed, everything is executed as expected. I guess `*_test` packages import their `*` package implicitly?

**Comment by slintes** [comment]
> Another questions:
> here we run tests with
> 
> `GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --keepGoing -requireSuite functests -- -junitDir /tmp/artifacts`
> 
> Will it catch also the new suite?

Thanks for the quick review!
Yes, the `-r` will make ginkgo executing all suites in `functests` recursively.

File: functests/utils/mcps/mcps.go (hunk 2 above)
Comment (slintes) [reply functests/utils/mcps/mcps.go:49]: Fair point, but actually I like stating explicitly that no namespace is needed. But there is a nicer way to do that, will change it to `metav1.NamespaceNone` (which is `""`, but nicer), which we use at other places as well.

File: functests/0_config/test_suite_performance_config_test.go (hunk 1 above)
Comment (fedepaol) [reply functests/0_config/test_suite_performance_config_test.go:12]: Ah yes, that's it. I did not notice they are in the same folder.

**Comment by fedepaol** [comment]
/lgtm
@slintes this needs rebase, will re-approve after

**Pull request merged**